metadata-ingestion/src/datahub/ingestion/source/aerospike.py

```diff
@@ -0,0 +1,591 @@
+            }
+
+        # initialize the schema for the set
+        canonical_schema: List[SchemaField] = []
+        max_schema_size = self.config.maxSchemaSize
+        set_schema_size = len(set_schema.values())
+        set_fields: Union[
+            List[SchemaDescription], ValuesView[SchemaDescription]
+        ] = set_schema.values()
+        assert max_schema_size is not None
+        if set_schema_size > max_schema_size:
+            # downsample the schema, using frequency as the sort key
+            self.report.report_warning(
+                title="Too many schema fields",
+                message=f"Downsampling the collection schema because it has too many schema fields. Configured threshold is {max_schema_size}",
+                context=f"Schema Size: {set_schema_size}, Collection: {dataset_urn}",
+            )
+            # Add this information to the custom properties so user can know they are looking at downsampled schema
+            dataset_properties.customProperties["schema.downsampled"] = "True"
+            dataset_properties.customProperties[
+                "schema.totalFields"
+            ] = f"{set_schema_size}"
+
+        logger.debug(f"Size of set fields = {len(set_fields)}")
+        # append each schema field (sort so output is consistent)
+        for schema_field in sorted(
+            set_fields,
+            key=lambda x: (
+                -x["count"],
+                x["delimited_name"],
+            ),  # Negate `count` for descending order, `delimited_name` stays the same for ascending
+        )[0:max_schema_size]:
+            field = SchemaField(
+                fieldPath=schema_field["delimited_name"],
+                nativeDataType=self.get_aerospike_type_string(
+                    schema_field["type"], dataset_urn.name
+                ),
+                type=self.get_field_type(schema_field["type"], dataset_urn.name),
+                description=None,
+                nullable=schema_field["nullable"],
+                recursive=False,
+            )
+            canonical_schema.append(field)
+
+        # create schema metadata object for set
+        return SchemaMetadata(
+            schemaName=as_set.set,
+            platform=dataset_urn.platform,
+            version=0,
+            hash="",
+            platformSchema=SchemalessClass(),
+            fields=canonical_schema,
+            primaryKeys=["PK"],
+        )
+
+    def xdr_sets(self, namespace: str, sets: List[str]) -> Dict[str, List[str]]:
+        sets_dc: Dict[str, List[str]] = {key: [] for key in sets}
+        shipped_sets = []
+        dcs = (
+            self.aerospike_client.info_random_node("get-config:context=xdr")
+            .split("dcs=")[1]
+            .split(";")[0]
+            .split(",")
+        )
+        if dcs == [""]:
+            logger.debug("No DCs found")
+            return sets_dc
+        for dc in dcs:
+            xdr_info: str = (
```

Comment:
minor: would recommend splitting into a readable sub-method: process_dc 

or whatever yu are doing in here..